python/triton/runtime/jit.py

```diff
@@ -686,13 +691,12 @@ def __init__(self, fn, version=None, do_not_specialize=None, do_not_specialize_o
     def get_capture_scope(self):
         return self.__globals__ | inspect.getclosurevars(self.fn).nonlocals
 
-    @property
-    def cache_key(self):
+    def get_cache_key(self, seen_fns=set()):
         # TODO : hash should be attribute of `self`
         if self.hash is None:
             nonlocals = inspect.getclosurevars(self.fn).nonlocals
             dependencies_finder = DependenciesFinder(name=self._fn_name, globals=self.__globals__, nonlocals=nonlocals,
-                                                     src=self.src)
+                                                     src=self.src, seen_fns=set(seen_fns))
```

Comment:
Could you avoid the changes to the signature by setting a temporary `self._in_dependency_finder| flag?